File: VB/CodeExamples/StylesActions.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native

Namespace RichEditDocumentServerAPIExample.CodeExamples

    Friend Class StylesAction

        Public Shared CreateNewCharacterStyleAction As System.Action(Of DevExpress.XtraRichEdit.RichEditDocumentServer) = AddressOf RichEditDocumentServerAPIExample.CodeExamples.StylesAction.CreateNewCharacterStyle

        Public Shared CreateNewParagraphStyleAction As System.Action(Of DevExpress.XtraRichEdit.RichEditDocumentServer) = AddressOf RichEditDocumentServerAPIExample.CodeExamples.StylesAction.CreateNewParagraphStyle

        Public Shared CreateNewLinkedStyleAction As System.Action(Of DevExpress.XtraRichEdit.RichEditDocumentServer) = AddressOf RichEditDocumentServerAPIExample.CodeExamples.StylesAction.CreateNewLinkedStyle

        Private Shared Sub CreateNewCharacterStyle(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#CreateNewCharacterStyle"
            ' Load a document from a file.
            wordProcessor.LoadDocument("Documents\Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            ' Access a document.
            Dim document As DevExpress.XtraRichEdit.API.Native.Document = wordProcessor.Document
            ' Access the character style with the specified name.
            Dim cstyle As DevExpress.XtraRichEdit.API.Native.CharacterStyle = document.CharacterStyles("MyCStyle")
            ' If the style with the specified name does not exist
            ' create a new character style and specify the style settings.
            If cstyle Is Nothing Then
                cstyle = document.CharacterStyles.CreateNew()
                cstyle.Name = "MyCStyle"
                cstyle.Parent = document.CharacterStyles("Default Paragraph Font")
                cstyle.ForeColor = System.Drawing.Color.DarkOrange
                cstyle.Strikeout = DevExpress.XtraRichEdit.API.Native.StrikeoutType.[Double]
                cstyle.FontName = "Verdana"
                ' Add the style to the collection of character styles.
                document.CharacterStyles.Add(cstyle)
            End If

            ' Access the range of the first paragraph.
            Dim myRange As DevExpress.XtraRichEdit.API.Native.DocumentRange = document.Paragraphs(CInt((0))).Range
            ' Access character formatting of the target range.
            Dim charProps As DevExpress.XtraRichEdit.API.Native.CharacterProperties = document.BeginUpdateCharacters(myRange)
            ' Apply the created character style to the target range.
            charProps.Style = cstyle
            ' Finalize to modify character formatting.
            document.EndUpdateCharacters(charProps)
#End Region  ' #CreateNewCharacterStyle
        End Sub

        Private Shared Sub CreateNewParagraphStyle(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#CreateNewParagraphStyle"
            ' Load a document from a file.
            wordProcessor.LoadDocument("Documents\Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            ' Access a document.
            Dim document As DevExpress.XtraRichEdit.API.Native.Document = wordProcessor.Document
            ' Access a paragraph style with the specified name.
            Dim pstyle As DevExpress.XtraRichEdit.API.Native.ParagraphStyle = document.ParagraphStyles("MyPStyle")
            ' If the style with the specified name does not exist
            ' create a new paragraph style and specify the style settings.
            If pstyle Is Nothing Then
                pstyle = document.ParagraphStyles.CreateNew()
                pstyle.Name = "MyPStyle"
                pstyle.LineSpacingType = DevExpress.XtraRichEdit.API.Native.ParagraphLineSpacing.[Double]
                pstyle.Alignment = DevExpress.XtraRichEdit.API.Native.ParagraphAlignment.Center
                ' Add the style to the collection of paragraph styles.
                document.ParagraphStyles.Add(pstyle)
            End If

            If document.Paragraphs.Count > 2 Then
                ' Apply the created paragraph style to the third document paragraph.
                document.Paragraphs(CInt((2))).Style = pstyle
            End If
#End Region  ' #CreateNewParagraphStyle
        End Sub

        Private Shared Sub CreateNewLinkedStyle(ByVal wordProcessor As DevExpress.XtraRichEdit.RichEditDocumentServer)
#Region "#CreateNewLinkedStyle"
            ' Access a document.
            Dim document As DevExpress.XtraRichEdit.API.Native.Document = wordProcessor.Document
            ' Start to edit the document.
            document.BeginUpdate()
            ' Append text to the document.
            document.AppendText("Line One" & Global.Microsoft.VisualBasic.Constants.vbLf & "Line Two" & Global.Microsoft.VisualBasic.Constants.vbLf & "Line Three")
            ' Finalize to edit the document.
            document.EndUpdate()
            ' Access a paragraph style with the specified name.
            Dim lstyle As DevExpress.XtraRichEdit.API.Native.ParagraphStyle = document.ParagraphStyles("MyLinkedStyle")
            ' If the style with the specified name does not exist
            ' create a new paragraph and character styles and specify their settings.
            If lstyle Is Nothing Then
                ' Start to edit the document.
                document.BeginUpdate()
                ' Create a paragraph style and specify its settings.
                lstyle = document.ParagraphStyles.CreateNew()
                lstyle.Name = "MyLinkedStyle"
                lstyle.LineSpacingType = DevExpress.XtraRichEdit.API.Native.ParagraphLineSpacing.[Double]
                lstyle.Alignment = DevExpress.XtraRichEdit.API.Native.ParagraphAlignment.Center
                document.ParagraphStyles.Add(lstyle)
                ' Create a character style and specify its settings.
                Dim lcstyle As DevExpress.XtraRichEdit.API.Native.CharacterStyle = document.CharacterStyles.CreateNew()
                lcstyle.Name = "MyLinkedCStyle"
                document.CharacterStyles.Add(lcstyle)
                ' Set the created character style to the created paragraph style.
                lcstyle.LinkedStyle = lstyle
                ' Specify the created character style's settings.
                lcstyle.ForeColor = System.Drawing.Color.DarkGreen
                lcstyle.Strikeout = DevExpress.XtraRichEdit.API.Native.StrikeoutType.[Single]
                lcstyle.FontSize = 24
                ' Finalize to edit the document.
                document.EndUpdate()
                ' Save the resulting document and select it in the File Explorer.
                document.SaveDocument("LinkedStyleSample.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
                System.Diagnostics.Process.Start("explorer.exe", "/select," & "LinkedStyleSample.docx")
            End If
#End Region  ' #CreateNewLinkedStyle
        End Sub
    End Class
End Namespace

File: CS/CodeExamples/StylesActions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;

namespace RichEditDocumentServerAPIExample.CodeExamples
{
    class StylesAction
    {
        public static Action<RichEditDocumentServer> CreateNewCharacterStyleAction = CreateNewCharacterStyle;
        public static Action<RichEditDocumentServer> CreateNewParagraphStyleAction = CreateNewParagraphStyle;
        public static Action<RichEditDocumentServer> CreateNewLinkedStyleAction = CreateNewLinkedStyle;

        static void CreateNewCharacterStyle(RichEditDocumentServer wordProcessor)
        {
            #region #CreateNewCharacterStyle
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents\\Grimm.docx", DocumentFormat.OpenXml);

            // Access a document.
            Document document = wordProcessor.Document;

            // Access the character style with the specified name.
            CharacterStyle cstyle = document.CharacterStyles["MyCStyle"];

            // If the style with the specified name does not exist
            // create a new character style and specify the style settings.
            if (cstyle == null)
            {
                cstyle = document.CharacterStyles.CreateNew();
                cstyle.Name = "MyCStyle";
                cstyle.Parent = document.CharacterStyles["Default Paragraph Font"];
                cstyle.ForeColor = System.Drawing.Color.DarkOrange;
                cstyle.Strikeout = StrikeoutType.Double;
                cstyle.FontName = "Verdana";
                // Add the style to the collection of character styles.
                document.CharacterStyles.Add(cstyle);
            }

            // Access the range of the first paragraph.
            DocumentRange myRange = document.Paragraphs[0].Range;
            
            // Access character formatting of the target range.
            CharacterProperties charProps =
                document.BeginUpdateCharacters(myRange);
            
            // Apply the created character style to the target range.
            charProps.Style = cstyle;

            // Finalize to modify character formatting.
            document.EndUpdateCharacters(charProps);
            #endregion #CreateNewCharacterStyle
        }

        static void CreateNewParagraphStyle(RichEditDocumentServer wordProcessor)
        {
            #region #CreateNewParagraphStyle
            // Load a document from a file.
            wordProcessor.LoadDocument("Documents\\Grimm.docx", DocumentFormat.OpenXml);

            // Access a document.
            Document document = wordProcessor.Document;

            // Access a paragraph style with the specified name.
            ParagraphStyle pstyle = document.ParagraphStyles["MyPStyle"];

            // If the style with the specified name does not exist
            // create a new paragraph style and specify the style settings.
            if (pstyle == null)
            {
                pstyle = document.ParagraphStyles.CreateNew();
                pstyle.Name = "MyPStyle";
                pstyle.LineSpacingType = ParagraphLineSpacing.Double;
                pstyle.Alignment = ParagraphAlignment.Center;
                // Add the style to the collection of paragraph styles.
                document.ParagraphStyles.Add(pstyle);
            }

            if (document.Paragraphs.Count > 2)
            {
                // Apply the created paragraph style to the third document paragraph.
                document.Paragraphs[2].Style = pstyle;
            }
            #endregion #CreateNewParagraphStyle
        }

        static void CreateNewLinkedStyle(RichEditDocumentServer wordProcessor)
        {
            #region #CreateNewLinkedStyle
            // Access a document.
            Document document = wordProcessor.Document;

            // Start to edit the document.
            document.BeginUpdate();

            // Append text to the document.
            document.AppendText("Line One\nLine Two\nLine Three");

            // Finalize to edit the document.
            document.EndUpdate();

            // Access a paragraph style with the specified name.
            ParagraphStyle lstyle = document.ParagraphStyles["MyLinkedStyle"];

            // If the style with the specified name does not exist
            // create a new paragraph and character styles and specify their settings.
            if (lstyle == null)
            {
                // Start to edit the document.
                document.BeginUpdate();

                // Create a paragraph style and specify its settings.
                lstyle = document.ParagraphStyles.CreateNew();
                lstyle.Name = "MyLinkedStyle";
                lstyle.LineSpacingType = ParagraphLineSpacing.Double;
                lstyle.Alignment = ParagraphAlignment.Center;
                document.ParagraphStyles.Add(lstyle);

                // Create a character style and specify its settings.
                CharacterStyle lcstyle = document.CharacterStyles.CreateNew();
                lcstyle.Name = "MyLinkedCStyle";
                document.CharacterStyles.Add(lcstyle);

                // Set the created character style to the created paragraph style.
                lcstyle.LinkedStyle = lstyle;

                // Specify the created character style's settings.
                lcstyle.ForeColor = System.Drawing.Color.DarkGreen;
                lcstyle.Strikeout = StrikeoutType.Single;
                lcstyle.FontSize = 24;

                // Finalize to edit the document.
                document.EndUpdate();

                // Save the resulting document and select it in the File Explorer.
                document.SaveDocument("LinkedStyleSample.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
                System.Diagnostics.Process.Start("explorer.exe", "/select," + "LinkedStyleSample.docx");
            }
            #endregion #CreateNewLinkedStyle
        }
    }
}
